### PR TITLE
completed assignment 1

### PR DESCRIPTION
**Pull request by jkhan2211**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I am adding new code to implement the KNN classification assignment. This includes data preprocessing, splitting into training/testing sets, hyperparameter tuning with grid search, and evaluating the final model using accuracy. The changes mainly involve adding new code cells and outputs rather than modifying or removing existing code.

## What did you learn from the changes you have made?
I learned how to properly preprocess and standardize data, tune hyperparameters with grid search, and evaluate a KNN classifier on a test set.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
N/A

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Understanding the concept, and coding the implementation, took a bit long to get hang of and understand, so I believe I would also need to try the extra activities to get a better grasp of the Classification, Knn neighbors implementation. 

## How were these changes tested?
By carefully examining the output, using print statements

## A reference to a related issue in your repository (if applicable)
N/A

## Checklist
- [X ] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by efantinatti** [comment]
Hi Junaid

# PR Feedback Report

**Student**: Junaid  
**GitHub Username**: @jkhan2211   
**PR Number**: 1  
**Branch**: assignment-1  
**Status**: INCOMPLETE  
**PR URL**: [#1](https://github.com/jkhan2211/LCR/pull/1)

## Overall Assessment

Incomplete, missing code and execution from questions 1 and 2.

**Comment by efantinatti** [comment]
Hi @jkhan2211 completed!

**Comment by jkhan2211** [comment]
> Hi Junaid
> 
> # PR Feedback Report
> **Student**: Junaid **GitHub Username**: @jkhan2211 **PR Number**: 1 **Branch**: assignment-1 **Status**: INCOMPLETE **PR URL**: [#1](https://github.com/jkhan2211/LCR/pull/1)
> 
> ## Overall Assessment
> Incomplete, missing code and execution from questions 1 and 2.

@efantinatti  - Hi Efanan, please mark as complete, we just discussed and you verified that the question 1 and 2 is complete :) Thank you

**Comment by jkhan2211** [comment]
> Hi @jkhan2211 completed!

@efantinatti - Can I go ahead and merge pull request or wait ? Thank you

**Comment by efantinatti** [comment]
Hi @jkhan2211 all good. You can merge it now. Completed.
